r/tests/testthat/test-Array.R

```diff
@@ -260,18 +260,28 @@ test_that("array supports POSIXct (ARROW-3340)", {
   expect_array_roundtrip(times2, timestamp("us", "US/Eastern"))
 })
 
-test_that("array supports POSIXct without timezone", {
-  # Make sure timezone is not set
+test_that("array uses local timezone for POSIXct without timezone", {
   withr::with_envvar(c(TZ = ""), {
     times <- strptime("2019-02-03 12:34:56", format = "%Y-%m-%d %H:%M:%S") + 1:10
-    expect_array_roundtrip(times, timestamp("us", ""))
+    expect_equal(attr(times, "tzone"), NULL)
+    expect_array_roundtrip(times, timestamp("us", Sys.timezone()))
 
     # Also test the INTSXP code path
     skip("Ingest_POSIXct only implemented for REALSXP")
     times_int <- as.integer(times)
     attributes(times_int) <- attributes(times)
     expect_array_roundtrip(times_int, timestamp("us", ""))
   })
+  withr::with_envvar(c(TZ = "Pacific/Marquesas"), {
+    times <- strptime("2019-02-03 12:34:56", format = "%Y-%m-%d %H:%M:%S") + 1:10
+    expect_equal(attr(times, "tzone"), "Pacific/Marquesas")
+    expect_array_roundtrip(times, timestamp("us", "Pacific/Marquesas"))
+  })
+  withr::with_envvar(c(TZ = NA), {
+    times <- strptime("2019-02-03 12:34:56", format = "%Y-%m-%d %H:%M:%S") + 1:10
+    expect_equal(attr(times, "tzone"), NULL)
+    expect_array_roundtrip(times, timestamp("us", Sys.timezone()))
+  })
```

Comment:
We might need to do something different here to confirm that these are actually having the effect we want:

```
> withr::with_envvar(list("TZ" = "Europe/London"), {
+   Sys.timezone()
+ })
[1] "Europe/London"
> 
> withr::with_envvar(list("TZ" = "Europe/Paris"), {
+   Sys.timezone()
+ })
[1] "Europe/Paris"
> 
> withr::with_envvar(list("TZ" = ""), {
+   Sys.timezone()
+ })
[1] "America/Chicago"
> 
> withr::with_envvar(list("TZ" = "Europe/Paris"), {
+   Sys.timezone()
+ })
[1] "America/Chicago"
```

We might want `withr::with_timezone` instead?